### PR TITLE
6 new field vehicle status 1

### LINKED ISSUE #6: New Field Vehicle Status
A requirement to introduce the field status to vehicles API PDC.
The possible status of the vehicle will be:

- INTRANS (Vehicle in transit)
- DLVRD (Vehicle has been delivered)
- INLOAD (Vehicle on a load)
- BUILT (Vehicle built)

### PR DESCRIPTION
**Pull request by pedro-acsousa**
Added the vehicle status after integrating with charlene. Tested and working with DB

### CODE REVIEW

File: src/main/java/com/jlr/ttl/ds/api/dto/entity/Vehicle.java [hunk 1]
```diff
@@ -17,4 +17,6 @@ public class Vehicle implements Entity<Vehicle> {
     private String vin;
     private String locCode;
     private Date endDate;
+
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/entity/Vehicle.java]: Remove empty line

File: src/main/java/com/jlr/ttl/ds/api/dto/response/VehicleResponse.java [hunk 2]
```diff
@@ -18,4 +18,6 @@ public class VehicleResponse implements DSResponseInterface<Vehicle> {
     private String vin;
     private String locCode;
     private Date endDate;
+
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/response/VehicleResponse.java]: Remove empty lines

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ public class VehicleService {
     public VehicleResponse getVehicleByID(String id) throws ServiceBusinessException {
         if(id==null || id.length()==0){
             String errorMessage = "No vehicle id was provided";
-            log.warn(errorMessage);
+            //log.warn(errorMessage);
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: Why did you comment the log

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public VehicleResponse getVehicleByID(String id) throws ServiceBusinessException
             return VehicleValueMapper.entityToResponse(dbResponse.createEntity());
         }catch (VehicleNotFoundException vehicleNotFoundException){
             String errorMessage = vehicleNotFoundException.getMessage();
-            log.warn(errorMessage);
+            //log.warn(errorMessage);
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: Why did you comment the log

**Review by BFARIAAL — CHANGES_REQUESTED** [review]
Needs to resolve comments. Other than that PR is correct

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java (hunk 4 above)
Comment (pedro-acsousa) [reply src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: This has been resolved

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java (hunk 3 above)
Comment (pedro-acsousa) [reply src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: This has been resolved

File: src/main/java/com/jlr/ttl/ds/api/dto/response/VehicleResponse.java (hunk 2 above)
Comment (pedro-acsousa) [reply src/main/java/com/jlr/ttl/ds/api/dto/response/VehicleResponse.java]: This has been resolved

File: src/main/java/com/jlr/ttl/ds/api/dto/entity/Vehicle.java (hunk 1 above)
Comment (pedro-acsousa) [reply src/main/java/com/jlr/ttl/ds/api/dto/entity/Vehicle.java]: This has been resolved

**Review by pedro-acsousa — COMMENTED** [review]
Resolved

**Review by BFARIAAL — APPROVED** [review]

**Pull request merged**